src/devices/miboxer.ts

```diff
@@ -23,16 +23,18 @@ const definitions: Definition[] = [
         configure: async (device, coordinatorEndpoint, logger) => {
             const endpoint = device.getEndpoint(1);
             await tuya.configureMagicPacket(device, coordinatorEndpoint, logger);
-            await endpoint.command('genGroups', 'miboxerSetZones', {zones: [
-                {zoneNum: 1, groupId: 101},
-                {zoneNum: 2, groupId: 102},
-                {zoneNum: 3, groupId: 103},
-                {zoneNum: 4, groupId: 104},
-                {zoneNum: 5, groupId: 105},
-                {zoneNum: 6, groupId: 106},
-                {zoneNum: 7, groupId: 107},
-                {zoneNum: 8, groupId: 108},
-            ]});
+
+            // Get unique identifier for the remote and convert it from hex-string to integer
+            const devId = parseInt(device.ieeeAddr, 16);
+
+            // Generate 8 globally unique (but reproducable) Zigbee group IDs
+            const groupIds = [devId+1, devId+2, devId+3, devId+4, devId+5, devId+6, devId+7, devId+8];
+
+            // Generate the zone mapping, which tells the remote to use the 8 unique group IDs for its 8 zones
+            const zoneToGroupMappings = groupIds.map((groupId, i) => ({zoneNum: i+1, groupId: groupId}));
+
+            // Send the zone mapping to the remote
+            await endpoint.command('genGroups', 'miboxerSetZones', {zones: zoneToGroupMappings});
```

Comment:
I'm not sure if I inderstand you correctly, can you elaborate? 

One thing I would be worried about (if we were to allow the users to set arbitrary group IDs) is that it wouldn't be possible to detect which button each group represents. With static groups 101-108 and even with my proposal, it would be possible to Identify the button (zone) and I kinda need that in order for this extension to work:  https://github.com/Koenkk/zigbee2mqtt-user-extensions/blob/main/unstable/miboxer-fut089z/controls-exposer.js

Ideally we wouldn't need any extensions at all, but that doesn't seem possible either.  :/

Reply:
> that it wouldn't be possible to detect which button each group represents

If you click the button, you can see the groupID in the Zigbee2MQTT debug log (right?)?

Reply:
Yes, I can see the groupID in the debug log:
```
debug 2023-10-11 14:09:26Received Zigbee message from 'MiBoxerRemote1', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 101
```
and currently I just know, that groupID 101 refers to the first on/off button group and 102 belongs to the second on/off button group. If the user would be able to specify for example id 999 for the first button group and 123 for the second button group, there would be no way to know which groupID belongs to which button group.

Reply:
> there would be no way to know which groupID belongs to which button group.

Can't you click the button and then in the z2m debug log see what group it belongs to?

Reply:
Yes, you can press a button on the remote and you can see which group it swnds a message to in the debug log. But there is no way to figure out which button was pressed fom the software side.

Reply:
> But there is no way to figure out which button was pressed fom the software side.

Can you provide me the debug log (including the `MQTT publish`) when a button is pressed?

Reply:
Sure:

```
Debug 2023-10-15 10:35:52Received Zigbee message from '0x003c84fffeb6a253', type 'commandOn', cluster 'genOnOff', data '{}' from endpoint 1 with groupID 107
Debug 2023-10-15 10:35:52No converter available for 'FUT089Z' with cluster 'genOnOff' and type 'commandOn' and data '{}'
```
There is no MQTT publish when pressing a button. Thats's why I wrote the extension. The extension just calculates `groupID-100` to figure out that it's the 7th button group and thus will do:
```
Info 2023-10-15 10:35:52MQTT publish: topic 'zigbee2mqtt/0x003c84fffeb6a253/action', payload 'button_group_7_on'
```

Reply:
Seems the converter is missing, I added it.

Could you check if the issue is fixed with the following external converter: https://gist.github.com/Koenkk/e0384c06554e330e8c94bb1fc1d68228

- save this as file next to `configuration.yaml` as `ext_converter.js`
- add it to `configuration.yaml`:
```yaml
external_converters:
  - ext_converter.js
```
- start z2m, check if issue is fixed